doc/design.adoc

```diff
@@ -0,0 +1,637 @@
+
+`InfinispanRemoteSiteSpec`.
+
+[options="header,footer"]
+|=======================
+| Field | Description | Scheme | Required | Default
+
+| `name`
+| Name of remote site.
+| string
+| true
+|
+
+| `type`
+| Type of remote site configuration.
+| `Static` or `Dynamic`
+| true
+|
+
+| `host`
+| Remote site host name.
+| string
+| true
+|
+
+| `port`
+| Remote site host port (only for `Static` type).
+| int32
+| false
+|
+
+| `secret`
+| Secret to connect to remote site (only for `Dynamic` type).
+| <<infinispansecretspec>>
+| false
+|
+
+|=======================
+
+
+[[infinispansecretspec]]
+##### `InfinispanSecretSpec`
+
+`InfinispanSecretSpec` defines how `InfinispanSpec` secrets are configured.
+
+[options="header,footer"]
+|=======================
+| Field | Description | Scheme | Required | Default
+
+| `type`
+| Type of secret.
+| `Credentials`, `Keystore` or `Token`
+| true
+|
+
+| `secretName`
+| Name of referenced secret.
+| string
+| true
+|
+
+|=======================
+
+If type is `Credentials`, Secret` is expected to contain username and password credentials.
+These would be defined in `stringData/username` and `stringData/password` fields respectively.
+
+If type is `Keystore`, `Secret` is expected to contain base64 encoded data in `data/keystore.p12` field.
+Optional keystore password would be located in `stringData/password` field.
+
+If type is `Token`, `Secret` is expected to contain base64 encoded data in `stringData/token` field.
+
+
+[[infinispanstatus]]
+### `InfinispanStatus`
+
+`InfinispanStatus` is the most recent observed status of the `InfinispanSpec`. Read-only.
+
+This is a proposal. TODO: Please comment
+
+[options="header,footer"]
+|=======================
+| Field | Description | Scheme | Required
+
+| `nodes`
+| Status of the nodes.
+| []<<InfinispanNodeStatus>>
+| true
+
+| `summary`
+| Short description of the cluster status. Now summary is `Ready` if all the pod conditions of type `Ready` are true 
+| string
+| true
+
+|=======================
+
+
+[[nodes]]
+#### `InfinispanNodeStatus`
+
+`InfinispanNodeStatus` is the most recent observed status of a pod running `InfinispanSpec`.
+
+[options="header,footer"]
+|=======================
+| Field | Description | Scheme | Required
+
+| `name`
+| Name of the Pod.
+| string
+| true
+
+| `clusterMembers`
+| List of cluster members (in Infinispan CLI format)
+| string
+| true
+
+| `clusterSize`
+| Number of cluster members
+| int
+| true
+
+|=======================
+
+## Full Example
+
+.full-example.yaml
+[source,yaml]
+----
+apiVersion: infinispan.org/v1
+kind: Infinispan
+metadata:
+  name: full-example-infinispan
+spec:
+  evictionPolicy: Reject
+  image: jboss/infinispan-server:latest
+  replicationFactory: 3
+  size: 4
+  cluster:
+    encryption:
+      type: Symmetric
+      secret:
+        type: Keystore
+        secretName: cluster-encrypt-secret
+    secret:
+      type: Credentials
+      secretName: cluster-auth-secret
+  connector:
+    secret:
```

Comment:
Good suggestion about `connector.authentication.secret` +1

For encryption, we should follow the same suggestion I made above: connector encryption is enabled by default transparently, with the operator doing what it needs to do to generate the encryption keys with OpenShift's certificate. Then, if using `dev` profile, this would be disabled.

Reply:
Is OpenShift provided endpoint encryption enough for users? I'm trying to think about the case when user would like to provide it's own keystore for the endpoint. (Keystore that would enable passthrough routes with REST as OpenShift doesn't provide certificates for routes but services)

Reply:
Hmmm, not sure I can picture an exact pattern or use case for this, but your suggestion around connector.encryption section can be added at a later stage once we have a more clear example. I've not heard a requirement for this yet.